### PR TITLE
Proposed table action group refactor

### PR DESCRIPTION
**Pull request by brmscheiner**
This proposed component would allow us to 
1. simplify creating tables with row actions
2. apply styles for these actions consistently across the board, and 
3. provide a way to implement these row actions cleanly without linter errors

thoughts?

### CODE REVIEW

File: src/pages/fieldManagement/settings/CustomFieldTable.jsx [hunk 1]
```diff
@@ -6,8 +6,8 @@ import Grid from '@material-ui/core/Grid';
 import Tooltip from '@material-ui/core/Tooltip';
 import AddIcon from '@material-ui/icons/Add';
 
+import { cellRendererTypes } from '../../../components/dataDisplays/cellRenderers';
```
Comment (brmscheiner) [inline src/pages/fieldManagement/settings/CustomFieldTable.jsx:9]: just one example refactor, next step would be to do it across the board

File: src/pages/fieldManagement/settings/CustomFieldTable.jsx [hunk 2]
```diff
@@ -62,6 +62,25 @@ export default function CustomFieldTable({
...
+  const deriveEditHref = useMemo(
+    () => (_, field) =>
+      `/settings/fields/save-custom-field/${fieldTypeName}/${field.id}`,
+    [fieldTypeName],
+  );
+
+  const onDeleteCustomField = useMemo(
+    () => (_, field) => setDeleteField(field),
+    [],
+  );
```
Comment (Emily-Ke) [inline src/pages/fieldManagement/settings/CustomFieldTable.jsx:82]: These can be simplified to `useCallback`s
> useCallback(fn, deps) is equivalent to useMemo(() => fn, deps)
https://reactjs.org/docs/hooks-reference.html#usecallback

Comment (brmscheiner) [reply src/pages/fieldManagement/settings/CustomFieldTable.jsx:82]: good point! i always forget about that one

**Comment by brmscheiner** [comment]
@Emily-Ke changes pushed!

**Pull request merged**